gms/gossiper.cc

```diff
@@ -2588,6 +2629,25 @@ inet_address_vector_replica_set gossiper::endpoint_filter(const sstring& local_r
     return result;
 }
 
+void gossiper::set_instance_id(const utils::UUID& instance_id) {
+    _instance_id = instance_id;
+    logger.info("Set instance_id to {}", _instance_id);
+}
+
+const utils::UUID& gossiper::get_instance_id() const {
+    return _instance_id;
+}
+
+void gossiper::add_blocked_instance_id(const utils::UUID& instance_id) {
+    _blocked_instance_ids.insert(instance_id);
+    logger.info("Add {} to blocked list, block_instance_ids={}", instance_id, _blocked_instance_ids);
+}
+
+bool gossiper::is_blocked_instance_id(const utils::UUID& instance_id) {
+    return _blocked_instance_ids.contains(instance_id);
```

Comment:
Maybe we should block all RPCs, and put this in messaging service. The safe topology changes algorithm needs this.